charmcraft/parts/charm.py

```diff
@@ -221,6 +221,7 @@ def get_build_packages(self) -> set[str]:
                 "python3-setuptools",
                 "python3-venv",
                 "python3-wheel",
+                "libyaml-dev",
```

Comment:
I think yum-based platforms would similarly benefit with `libyaml-devel`, right?

Reply:
Most likely, yes. For apt, I'm familiar enough with the platforms (well, Debian and Ubuntu at least) that I'm confident that this will be ok, and I tested it manually with Ubuntu, with and without the build package and verifying the behaviour in each case. For other platforms, I'm pretty sure that a similar solution would help but I wasn't certain and didn't take the time to test it on those kinds images.

Reply:
+1 on keeping deps consistent across the various bases, if at all possible.

Reply:
I've opened #1771 for this, but I'm not going to hold up the merge on that.